docs/standard/commandline/tab-completion.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Tab completion for System.CommandLine
+description: "How to enable and customize tab completion for apps built on the System.CommandLine library."
+ms.date: 02/03/2022
+no-loc: [System.CommandLine]
+helpviewer_keywords:
+  - "command line interface"
+  - "command line"
+  - "System.CommandLine"
+ms.topic: how-to
+---
+
+# Tab completion for System.CommandLine
+
+Apps that use `System.CommandLine` have built-in support for tab completion in certain shells. To enable it, the end user has to take a few steps once per shell. Tab completion is automatic for static values in your app, such as enum values or values you define by calling [FromAmong](define-commands.md#list-valid-argument-values). You can also customize the tab completion by getting values dynamically at runtime.
+
+## Enable tab completion
+
+On the machine where you'd like to enable tab completion, do the following steps.
+
+For the .NET CLI:
+
+* See [How to enable tab completion](../../core/tools/enable-tab-autocomplete.md).
+
+For other command-line apps built on `System.CommandLine`:
+
+* Install the [`dotnet-suggest`](https://nuget.org/packages/dotnet-suggest) global tool.
+
+* Add the appropriate shim script to your shell profile. You may have to create a shell profile file. The shim script forwards completion requests from your shell to the `dotnet-suggest` tool, which delegates to the appropriate `System.CommandLine`-based app.
+
+  * For `bash`, add the contents of [*dotnet-suggest-shim.bash*](https://github.com/dotnet/command-line-api/blob/main/src/System.CommandLine.Suggest/dotnet-suggest-shim.bash) to *~/.bash_profile*.
+
+  * For PowerShell, add the contents of [*dotnet-suggest-shim.ps1*](https://github.com/dotnet/command-line-api/blob/master/src/System.CommandLine.Suggest/dotnet-suggest-shim.ps1) to your PowerShell profile. You can find the expected path to your PowerShell profile by running the following command in your console:
+
+    ```console
+    echo $profile
+    ```
+
+Once the user's shell is set up, completions will work for all apps that are built by using `System.CommandLine`.
+
+For *cmd.exe* on Windows there is no pluggable tab completion mechanism, so no shim script is available. For other shells, [look for a GitHub issue that is labeled `Area-Completions`](https://github.com/dotnet/command-line-api/issues?q=is%3Aissue+is%3Aopen+label%3A%22Area-Completions%22). If you don't find an issue, you can [open a new one](https://github.com/dotnet/command-line-api/issues).
```

Comment:
This is a very helpful section - I would suggest also calling out the shim script for `zsh`, because it's the default shell for MacOS these days.